src/query/storage/cache/redis_cache_test.go

```diff
@@ -0,0 +1,234 @@
+// Simple unit tests for Redis cache
+// This test suite requires setting up a Redis instance at
+// 127.0.0.1:6379 (localhost)
+
+package cache
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/m3db/m3/src/query/block"
+	"github.com/m3db/m3/src/query/generated/proto/prompb"
+	"github.com/m3db/m3/src/query/models"
+	"github.com/m3db/m3/src/query/storage"
+	radix "github.com/mediocregopher/radix/v3"
+	"github.com/stretchr/testify/require"
+	"github.com/uber-go/tally"
+	"go.uber.org/zap"
+)
+
+var (
+	cache RedisCache
+)
+
+func TestMain(m *testing.M) {
```

Comment:
Can we add unit test to cover WindowGetOrFetch method with some corner cases?

Reply:
So these unit tests are run locally without making any connection to an M3DB instance and making a connection to one might be a bit tricky so I'm not sure if we can implement this in these unit tests, will work to add more corner cases

Reply:
Update: found a way to test it without M3DB by basing it off some predefined unit tests in M3 already